inc/planningcsv.class.php

```diff
@@ -0,0 +1,187 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * Planning CSV export Class
+**/
+class PlanningCsv extends CommonGLPI {
+
+   private $users_id;
+   private $groups_id;
+   private $limititemtype;
+
+   private $eol = "\r\n";
+   private $quote = '"';
+   private $titles = [
+      'actor',
+      'title',
+      'itemtype',
+      'items_id',
+      'begindate',
+      'enddate'
+   ];
+   private $lines = null;
+   private $filename = 'planning.csv';
+
+   /**
+    * @param integer $who          user ID
+    * @param integer $who_group    group ID
+    * @param string $limititemtype itemtype only display this itemtype (default '')
+   **/
+
+   public function __construct($who, $who_group = null, $limititemtype = '') {
+      $this->users_id = $who;
+      $this->groups_id = $who_group;
+      $this->limititemtype = $limititemtype;
+   }
+
+   /**
+    * Build CSV lines
+    *
+    * @return void
+    */
+   private function buildLines() {
+      global $CFG_GLPI;
+
+      $interv = [];
+      $this->lines = [];
+      $begin  = time()-MONTH_TIMESTAMP*12;
+      $end    = time()+MONTH_TIMESTAMP*12;
+      $begin  = date("Y-m-d H:i:s", $begin);
+      $end    = date("Y-m-d H:i:s", $end);
+      $params = [
+         'who'       => $this->users_id,
+         'who_group' => $this->groups_id,
+         'whogroup'  => $this->groups_id,
+         'begin'     => $begin,
+         'end'       => $end
+      ];
+
+      if (empty($this->limititemtype)) {
+         foreach ($CFG_GLPI['planning_types'] as $itemtype) {
+            $interv = array_merge($interv, $itemtype::populatePlanning($params));
+         }
+      } else {
+         $interv = $this->limititemtype::populatePlanning($params);
+      }
+
+      if (count($interv) > 0) {
+         foreach ($interv as $val) {
+            $dateBegin = new DateTime($val["begin"]);
+            $dateBegin->setTimeZone(new DateTimeZone('UTC'));
+
+            $dateEnd = new DateTime($val["end"]);
+            $dateEnd->setTimeZone(new DateTimeZone('UTC'));
+
+            $itemtype = new $val['itemtype'];
+            //(acteur;titre item;id item;date-heure début,date-heure fin;catégorie)
+            $this->lines[] = [
+               'actor'     => $val['users_id'],
```

Comment:
I would export name of user using `User::getRawName()` or something human readable..